backend/corpora/lambdas/api/v1/curation/collections/collection_id/actions.py

```diff
@@ -36,4 +40,12 @@ def get(collection_id: str, token_info: dict):
 
 
 def patch(collection_id: str, body: dict, token_info: dict):
-    return update_collection(collection_id, body, token_info)
+    if "links" in body:
+        if not body["links"]:
+            raise InvalidParametersHTTPException(detail="If provided, the 'links' array may not be empty")
+        keep_links = False  # links have been provided; replace all old links
+    else:
+        keep_links = True  # links have NOT been provided; keep existing links
+    collection = update_collection_common(collection_id, body, token_info, keep_links=keep_links)
+    columns_to_return = ("name", "description", "contact_name", "contact_email", "links")
```

Comment:
@Bento007 just out of curiosity, is there a way to dynamically inspect the Response fields of the API definition here? Tried looking up an answer to this and didn't have much luck.

Not necessarily but could be nice, would be a bit cleaner and more maintainable.

Reply:
Not an easy way that I know of. You'd need to dig deep into the connexions code, or parse the swagger yourself. I think parsing the swagger would be the easy option, but I wouldn't recommend it. Personally I'd like to move away from connexion, and use something like fast-api, or [flask + marshmallow](https://flask-marshmallow.readthedocs.io/en/latest/).

Reply:
There is even a [flask+pydantic](https://pypi.org/project/flask-pydantic-spec/)